### DISCUSSION TITLE
Welcome to Type Challenges Discussions!

### DISCUSSION THREAD

**Discussion by antfu** [type-challenges/type-challenges#460] [General]
## 👋 Welcome!

  We’re using Discussions as a place to connect with other members of our community. We hope that you:

  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Join our [Discord](https://discord.gg/UgKBCq9) as well!

**Comment by johnsonjo4531** [comment]
Hey Anthony! I guess I'll introduce myself my name is John Johnson. I graduated in CS at USU about two years ago coming up on three, though I've been in the workforce programming for over eight years. I enjoy programming in my free time sometimes, but also like math (though I don't profess to be excellent at it), and reading (mostly non-fiction). I started using typescript about 2-3 years ago. I enjoy it so far, but I come to this community to learn more about type manipulation. It seems sometimes I take DRY a bit too seriously, and try not to repeat my types always using few types to explain something that could possibly take more. I feel too often I take myself down a rabbit hole of sorts with types that just never seems to work quite the way I would like, and I am all about providing the best possible type completions I can get for my team. This repo and collection of type challenges has already given me a much greater amount of understanding for when I have to do some crazy type manipulations. Anyways, I appreciate the community you're building Anthony. I originally joined this community I believe, because I saw it on Hacker News (https://news.ycombinator.com).

**Comment by antfu** [comment]
@johnsonjo4531 Thanks a lot for the feedback and glad you get something to take away from this project! I personally ran into a similar situation that trying to provide a complex typing check for my team and coworkers. I also found that solving them is a quite challenging but pleasant task. Then I am thinking, why not just collect them so people can play with and have a reference to? Which I believe you already knew the outcome :P

I wasn't aware we were on Hacker News, that's good info to me, thanks!

**Reply by johnsonjo4531** [reply]
Yeah the post on HN garnered quite a few points about 26 (https://news.ycombinator.com/item?id=24407005) and I think it made it on the second page atleast. Honestly I'm suprised it hadn't gotten more points given how excellent of a resource is, and it also gained about 6 comments. I spend way too much time on HN :P.

Thanks again for your awesome work! I'll continue practicing my types through here.

**Comment by makeevvd** [comment]
When I try no run "npm i", I get " Unsupported URL Type "workspace:": workspace:*". When I try "pnpm install", I get "Cannot find module server.js". 

Could you please help?

**Reply by antfu** [reply]
You would need to use pnpm for this project. https://pnpm.js.org/
